.github/workflows/tests.yml

```diff
@@ -1,38 +1,66 @@
-# This workflow performs static analysis checks.
-name: tests
+# This workflow does static verification of the DafnyCore library
+name: Dafny Core verification
 
 on:
+  # Scheduled to be run sometime after the nightly build of dafny
+  schedule:
+    - cron: "7 10 * * *"
   workflow_dispatch:
-  push:
-    branches: [ master ]
   pull_request:
     branches: [ master ]
 
 jobs:
   verification:
-    runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      fail-fast: false
+      matrix:
+        # nightly-latest to catch anything that breaks these tests in current development
+        # 2/28/2023 version is the first that supports logging
+        # 3.11.0 supports new CLI but does not support logging
+        version: [ nightly-latest, nightly-2023-02-18-ef4f346, 3.11.0]
+        os: [ ubuntu-latest ]
+
+    runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
       - name: Install Dafny
-        uses: dafny-lang/setup-dafny-action@v1
+        uses: dafny-lang/setup-dafny-action@v1.6.0
         with:
-          dafny-version: "3.9.1"
+          dafny-version: ${{ matrix.version }}
+
+      - name: Version information
+        run: |
+          dafny --version
+          echo ${{ matrix.os }} ${{ runner.os }} ${{ matrix.version }}
+
+      - name: Upgrade outdated pip
+        run: python -m pip install --upgrade pip
 
       - name: Install lit
         run: pip install lit OutputCheck
 
       - name: Set up JS dependencies
         run: npm install bignumber.js
 
-      - name: Verify Dafny Code
-        run: lit --time-tests -v --param 'dafny_params=/verificationLogger:trx /verificationLogger:csv' .
+      - name: Verify Code and Examples without logging
+        id: nolog
+        if: matrix.version == '3.11.0'
+        run: lit --time-tests -v .
+
+      - name: Verify Code and Examples
+        id: withlog
+        if: steps.nolog.conclusion == 'skipped'
+        run: |
+          lit --time-tests -v --param 'dafny_params=--log-format trx --log-format csv' .
 
       - name: Generate Report
+        if: always() && steps.withlog.conclusion != 'skipped'
         run: find . -name '*.csv' -print0 | xargs -0 --verbose dafny-reportgenerator summarize-csv-results --max-duration-seconds 10
 
       - uses: actions/upload-artifact@v2  # upload test results
-        if: success() || failure()        # run this step even if previous step failed
+        if: always() && steps.withlog.conclusion != 'skipped'
```

Comment:
So that the step runs even if the previous steps fail

Reply:
Then why not JUST always()?

Reply:
Because if we (intentionally) run with an old version of dafny that does not support logging with new CLI (i.e. 3.11.0) then we don't want this step to run (and fail) if we have not produced any log files.